examples/do.kl

```diff
@@ -1,8 +1,9 @@
 #lang "prelude.kl"
 
 (import (shift "prelude.kl" 1))
-(import (shift "quasiquote.kl" 1))
-(import (shift "let.kl" 1))
+(import (shift "dot-dot-dot.kl" 1))
+(import (shift "do-keywords.kl" 1))
+(import "do-keywords.kl")
```

Comment:
These last two imports are a symptom of something really sad about `fancy-syntax-case`: keywords must be visible both at the macro-definition phase and at the macro-caller's phase. This can either be done by defining them twice (once with `meta`) or by defining them in a separate module and them importing that module twice. Either way, it's really annoying.

The reason they need to be available in the macro-definition phase is because `fancy-syntax-case` uses `free-identifier=?` to check that every keyword used in the patterns have been listed in the keyword list. But if the keyword isn't in scope, then the occurrence of the keyword in the keyword list is not `free-identifier=?` to the occurrence in the pattern, resulting in spurious `did you mean to add the symbol to the keyword list?` errors.

The reason they need to be available in the macro-caller's phase is because the generated code uses `free-identifier=?` to compare the input syntax object against the various patterns, including against keywords. So keywords must be in scope in the phase in which the generated code runs or they won't be `free-identifier=?`.

One solution could be to drop that keyword list, and to trust that `fancy-syntax-case`'s caller doesn't accidentally write `[(_ cond then-expr else-expr) ...]` instead of `[(_ ,cond ,then-expr ,else-expr) ...]`. But since this is an unorthodox syntax, I expect that mistake to be quite common!

My preferred solution would be to add a new primitive, `define-keyword`, which would bind an identifier at _all_ phases. It would hardcode the value to which it is bound to a macro complaining that the keyword is used out of context, and thus we won't have to worry about things like whether the closure of the definition includes values which only exist at some phases. Would that make sense?